ls1intum/Artemis
# [Exam] Exam Management: Students

### Checklist

### Motivation and Context
This PR adds the client-side component for the students view for an exam. This view is used to add and remove students to and from and exam.

### Description
- Add component `exam-students`
- Use exercise management service methods for adding and removing a student from an exam
- Add option `withStudents` to `getExam` endpoint to fetch students with the exam
- Add component `students-exam-import-dialog`
- Students can be bulk imported into an exam via a csv file

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Management
3. Click on Exams for a Course (This button is hidden, go to `course-management/{courseId}/exams`)
4. Create an Exam if needed
5. Click on Students for an Exam
6. Search for a student to add them to the Exam
7. Click on "Remove" to remove them from the Exam
8. Click on "Import students" and select a csv file _(see sample in the screenshot section)_
9. Click on "Import" and see if the import works as expected

### Mockup
![84212970-b4fa4f00-aabf-11ea-951d-db3ed3ad5b5c](https://user-images.githubusercontent.com/7109225/84823420-97c50380-b01e-11ea-879b-f154828fa04f.png)

### Screenshots
![OnPaste 20200616-215738](https://user-images.githubusercontent.com/7109225/84822191-9561aa00-b01c-11ea-9218-53c64ac1b060.png)

#### Adding and removing a student
![OnPaste 20200616-215840](https://user-images.githubusercontent.com/7109225/84822194-9692d700-b01c-11ea-89c4-b649018c8372.png)
![OnPaste 20200616-215801](https://user-images.githubusercontent.com/7109225/84822192-95fa4080-b01c-11ea-96ad-a8e6b068fac5.png)
![OnPaste 20200616-215823](https://user-images.githubusercontent.com/7109225/84822193-9692d700-b01c-11ea-929e-ce3d9ad8b644.png)

#### Importing students
Sample file (needs to be renamed to .csv after download, Github doesn't allow .csv files) [students.txt](https://github.com/ls1intum/Artemis/files/4793319/students.txt)

![OnPaste 20200617-165531](https://user-images.githubusercontent.com/7109225/84916124-f42b3000-b0bd-11ea-9d32-33e635841042.png)
![OnPaste 20200617-165614](https://user-images.githubusercontent.com/7109225/84916127-f55c5d00-b0bd-11ea-9dbb-0bfd3a94dfc8.png)
![OnPaste 20200617-175839](https://user-images.githubusercontent.com/7109225/84921022-38b9ca00-b0c4-11ea-9c51-aeabdf052b93.png)

##### Validation
![OnPaste 20200617-223914](https://user-images.githubusercontent.com/7109225/84948105-7ed65400-b0eb-11ea-98be-e5c0da6e6aaf.png)
